subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/FilesFinder.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache.internal;
+
+import java.io.File;
+import java.io.FileFilter;
+
+/**
+ * Encapsulates a criteria for finding files.
+ */
+public interface FilesFinder {
+    /**
+     * Find files according to this finder's criteria within the supplied base
+     * directory that pass the supplied {@link FileFilter}.
+     */
+    File[] find(File baseDir, FileFilter filter);
```

Comment:
Originally, the most used implementation used `File.listFiles()`. But since, I've since removed that implementation I'll change it.

- [ ] Return `Collection<File>`